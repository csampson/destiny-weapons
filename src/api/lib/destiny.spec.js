/* eslint-env jest */

const axios = require('axios')
const MockAdapter = require('axios-mock-adapter')
const mockAxios = new MockAdapter(axios)

const Destiny = require('./destiny')

function mock (url) {
  mockAxios.onGet(RegExp(url)).reply(config => {
    expect(config.headers['X-Api-Key']).toMatchSnapshot()
    expect(config.baseURL).toMatchSnapshot()
    expect(config.url).toMatchSnapshot()

    return [200, {}]
  })
}

describe('Destiny', () => {
  afterEach(() => {
    mockAxios.reset()
  })

  describe('Armory', () => {
    describe('::search', () => {
      it('should search for armory items', async () => {
        mock('Armory/Search/.*')
        await Destiny.Armory.search('<type>', '<query>')
      })
    })
  })

  describe('Manifest', () => {
    describe('::fetch', () => {
      it('should fetch the manifest for a given entity', async () => {
        mock('Manifest/.*')
        await Destiny.Manifest.fetch('<type>', '<id>')
      })
    })
  })
})
