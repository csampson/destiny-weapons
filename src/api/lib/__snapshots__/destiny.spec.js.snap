// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Destiny Armory ::search should search for armory items 1`] = `"xxxx"`;

exports[`Destiny Armory ::search should search for armory items 2`] = `"https://www.bungie.net/Platform/Destiny2"`;

exports[`Destiny Armory ::search should search for armory items 3`] = `"/Armory/Search/<type>/<query>"`;

exports[`Destiny Manifest ::fetch should fetch the manifest for a given entity 1`] = `"xxxx"`;

exports[`Destiny Manifest ::fetch should fetch the manifest for a given entity 2`] = `"https://www.bungie.net/Platform/Destiny2"`;

exports[`Destiny Manifest ::fetch should fetch the manifest for a given entity 3`] = `"/Manifest/<type>/<id>"`;
